caveclient/base.py

```diff
@@ -251,7 +251,10 @@ def server_version(self) -> Optional[Version]:
         """The version of the service running on the remote server. Note that this
         refers to the software running on the server and has nothing to do with the
         version of the datastack itself."""
-        return self._server_version
+        if self._server_version is None and self._api_version is not None:
```

Comment:
@fcollman as an example, this logic would set some defaults for the versioning logic/decorator that uses the current _api_version to tell if a server is at least 2.x.y, 3.x.y, etc. 

I think this would make things more backwards compatible generally, but I wonder if you see any holes in this logic. 

Version("3") essentially resolves to 3.0.0, so any checks that look like >=3.1.2 etc would still fail (I just checked this)